src/RippleKey.h

```diff
@@ -0,0 +1,134 @@
+//------------------------------------------------------------------------------
+/*
+This file is part of ripple-sign-serialize:
+https://github.com/ximinez/ripple-sign-serialize
+Copyright (c) 2017 Ripple Labs Inc.
+
+Permission to use, copy, modify, and/or distribute this software for any
+purpose  with  or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/st.h>
+
+namespace boost
+{
+namespace filesystem
+{
+class path;
+}
+}
+
+namespace serialize {
+
+class RippleKey
+{
+private:
+    static ripple::KeyType constexpr defaultKeyType()
+    {
+        return ripple::KeyType::secp256k1;
+    }
+    ripple::KeyType keyType_;
+    ripple::Seed seed_;
+    ripple::PublicKey publicKey_;
+    ripple::SecretKey secretKey_;
+
+public:
+    RippleKey()
+        : RippleKey(RippleKey::defaultKeyType())
+    {}
+
+    explicit
+    RippleKey(ripple::KeyType const& keyType)
+        : RippleKey(keyType, ripple::randomSeed())
+    {}
+
+
+    RippleKey(
+        ripple::KeyType const& keyType,
+        ripple::Seed const& seed);
+
+    /** Attempt to construct RippleKey by parsing a seed
+
+        @param keyType Key type
+        @param rawseed String containing the seed
+
+        @throws std::runtime_error if the `rawseed` can not be parsed
+                into a `Seed`
+    */
+    static
+    RippleKey
+    make_RippleKey(ripple::KeyType const& keyType,
+        std::string rawseed);
+
+    /** Attempt to construct RippleKey with variable parameters
+
+        @param keyType Optional key type
+        @param rawseed Optional string containing the seed
+
+        @throws std::runtime_error if the `rawseed` is set and can not
+            be parsed into a `Seed`
+    */
+    static
+    RippleKey
+    make_RippleKey(boost::optional<ripple::KeyType> const& keyType,
+        boost::optional<std::string> const& rawseed);
+
+    /** Returns RippleKey constructed from JSON file
+
+        @param keyFile Path to JSON key file
+
+        @throws std::runtime_error if file content is invalid
+    */
+    static
+    RippleKey
+    make_RippleKey(
```

Comment:
The three ctors are written to handle or hide default behavior, particularly the one that gets a `randomSeed` if a seed isn't provided.  I was able to inline one of the overrides of `make_RippleKey()`, but the other two do completely different things (reading from a file vs taking parameters), so I left those alone.